test/unit/signup/account_manager_test.rb

```diff
@@ -126,23 +126,29 @@ class ProviderAccountManagerTest < Signup::AccountManagerTest
       assert_equal 'API', account.first_service!.name
     end
 
-    test 'creating a provider will create default Backend if api_as_product is disabled' do
+    test 'creating a provider will create default Backend regardless of api_as_product' do
       Account.any_instance.stubs(:provider_can_use?).returns(false)
       Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false)
       account = signup_account_manager.create(signup_params).account
-      assert_equal BackendApi.default_api_backend, account.default_service.api_backend
+      assert_expected_backend_api(account)
```

Comment:
The reason why it was kept the assertion of `account.default_service.api_backend` here is to abstract implementation details. For non-APIAP scenario. the seeds do not care about existing backend_apis.

```suggestion
      assert_equal BackendApi.default_api_backend, account.default_service.api_backend
```

Reply:
@guicassolato I did not remove that assertion at all:
https://github.com/3scale/porta/pull/1294/files#diff-e76abd3a417934b609ad05f2a208fa52R149
The goal of putting it in a method was just to don't repeat all the same assertions for both tests.

Reply:
https://github.com/3scale/porta/pull/1294#discussion_r330980774